specs/version_format_specification.md

```diff
@@ -103,6 +105,14 @@ Examples (with '' meaning the empty string):
 * `0.0` > `0`
 * `0` < `~`
 * '' < `~`
+* `1_` == `1`
+* `_1` == `1`
+* `1_` < `1.2`
+* `1_2_3` > `1.3.3`
```

Comment:
I would add an explanation here:
"The underscore is a separator between components, so we first compare "1" with "1.3.3" as numerical version strings, and 1 < 1.3.3. The remainder of the first string is not used in the comparison."